test/specialfunctionstests.jl

```diff
@@ -0,0 +1,16 @@
+using FastTransforms
+using Compat.Test
+import FastTransforms.pochhammer
+
+const sqrtpi = 1.772453850905516027298
+
+@testset "Pochhammer" begin
+    @test pochhammer(2,3) == 24
+    @test pochhammer(0.5,3) == 0.5*1.5*2.5
+    @test pochhammer(0.5,0.5) == 1/sqrtpi
+    @test pochhammer(0,1) == 0
+    @test pochhammer(-1,2) == 0
+    @test pochhammer(-5,3) == -60
+    @test pochhammer(-1,-0.5) == 0
+    @test 1.0/pochhammer(-0.5,-0.5) == 0
```

Comment:
Add complex tests?

Reply:
`gamma` is nonzero and holomorphic except the singularities, so complex tests don't make any difference, I suppose. I skipped array tests, though. Maybe add some array tests

Reply:
I mean complex types  but at integers: ` ogamma(-1+0im) `